script.js

```diff
@@ -0,0 +1,149 @@
+
+const APIKey= 'ef60cd7a93d72315d97d352ed875d11c'
+const baseURL= 'https://api.themoviedb.org/3/movie/now_playing'
+const baseSearchURL= 'https://api.themoviedb.org/3/search/movie'
+
+const state= {
+    searchTerm: '',
+    page: 1,
+    searching: false
+}
+let url= `${baseURL}?page=${state.page}&api_key=${APIKey}`
+
+
+const searchForm= document.getElementById('search-form')
+const searchInput= document.getElementById('search-input')
+const movie_container= document.getElementById('movies-grid')
+const showMoreBtn= document.getElementById('load-more-movies-btn')
+const closeSearchBtn= document.getElementById('close-search-btn')
+
+
+function displayResults(results){
+    //updating DOM
+    for (const movie of results){
+        generateCards(movie)
+    }
+}
+
+
+
+/**
+ * 
+ * @param {string} searchTerm 
+ * @param {boolean} initial 
+ * @returns data requested from API 
+ */
+async function getMovieAPIRes(searchTerm, page, searching=false){
+    if (searching){
+        url= `${baseSearchURL}?query=${searchTerm}&page=${page}&api_key=${APIKey}`
+    } else {
+        url= `${baseURL}?page=${page}&api_key=${APIKey}`
+    }
+    console.log(url)
+    const data= fetch(url)
+        .then((response) => response.json())
+    const result= await data
+    return result
+}   
+
+
+function generateCards(movieObj){
+    // create star
+    let star= document.createElement('span')
+    star.classList.add('star')
+    let starContent= document.createTextNode('⭐️')
+    star.appendChild(starContent)
+    document.body.appendChild(star)
+
+
+    // create rating
+    let avg= movieObj.vote_average
+    let rating= document.createElement('span')
+    rating.classList.add('avg')
+    let ratingContent= document.createTextNode(avg)
+    rating.appendChild(ratingContent)
+    document.body.appendChild(rating)
+
```

Comment:
recommend using a helper function to create/append the new elements (makes code more DRY)